src/fw/drivers/imu/lsm6dso/lsm6dso.c

```diff
@@ -78,6 +78,62 @@ lsm6dso_state_t s_lsm6dso_state = {0};
 lsm6dso_state_t s_lsm6dso_state_target = {0};
 static bool s_interrupts_pending = false;
 static uint32_t s_tap_threshold = BOARD_CONFIG_ACCEL.accel_config.double_tap_threshold / 1250;
+static bool s_fifo_in_use = false; // true when we have enabled FIFO batching
+
+// Maximum FIFO watermark supported by hardware (diff_fifo is 10 bits -> 0..1023)
+#define LSM6DSO_FIFO_MAX_WATERMARK 1023
+
+// Map output data rate (interval) to FIFO batching rate enum
+static lsm6dso_bdr_xl_t prv_get_fifo_batch_rate(uint32_t interval_us) {
```

Comment:
nit: I was pretty careful in my PR to group functions based on their associated interface implementation or functionality; with private methods below all interface methods. This doesn't conform with that convention; fwiw.